docs/Caching.md

```diff
@@ -0,0 +1,158 @@
+still require the browser application to make an HTTP API request for each
+active query, but the response would be delivered faster in some cases because
+the app server would not have to generate a corresponding HTTP request to the
+REST API server.
+
+2. **CDN** - cache batched API responses from app server
+
+    The next level of caching could appear between the CDN and the browser
+application. If the app server sent batched responses with a `Cache-Control`
+header that matched the _minimum cache time_ of the individual API responses,
+the CDN could be configured to cache those batched responses. For example, if
+a request was made for a group event, the batched response would contain
+responses from the _group endpoint_ and the _event endpoint_. The app server
+could apply a `Cache-Control` header that matched the shortest cache
+time-to-live of those two responses, and the CDN could cache that combined
+response and deliver it to _every user who subsequently made the same batched
+request_. This cache behavior would not be useful if every batched request
+included an un-cacheable response (e.g. a private/logged-in response, like 
+`/members/self`), so it would have to be developed after the app was updated to
+cache individual API responses and only request stale data, as described below.
+
+3. **Redux middleware** - cache parsed API responses and filter `fetch`
+
+    The next level of caching would require changes to the query-based API proxy
+routine. Each individual query response would need to contain the cache control
+rules of the corresponding REST API response so that the browser could
+appropriately cache the responses _and avoid making the same query request_
+while the browser-cached response was available. A complintary caching stategy
+could be applied at the CDN layer - if the CDN was configured to cache responses
+from the `/mu_api` endpoint based on
+
+    Although a service worker might be able to provide a 'transparent' caching
+solution, the batched requests and responses would be awkward to parse in order
+to correctly refine them to include only stale data.
+
+    Instead, cache control rules should probably be handled by a more sophisticated 
+interaction between the Sync epic (middleware) and the Cache epic
+to allow `fetch` requests to be modified (or cancelled) when cached data was
+available for the requested queries. We would probably still want to use the
+Cache epic to cache everything so that the UI was responsive even when 'fresh'
+data was being fetched, but the cache control rules would allow the `fetch` to
+be fine-tuned to include only the data that was known to _require_ an update.
+
+## 2. Static assets
+
+Static assets generated by the build are cached in our CDN and in the browser.
+Because each filename includes a hash of the file contents, the file can have an
+infinite cache lifetime.
+
+### CDN
+
+(CDN cache invalidation policy?)
```

Comment:
Thinking about this a bit more, I think we actually don't have to worry about invalidating the fastly cache since it only uses hashed filenames - the cache doesn't need to expire, so I've added that statement